frontend/src/Components/forms/LoginForm.tsx

```diff
@@ -1,93 +1,137 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useForm, SubmitHandler } from 'react-hook-form';
 import Checkbox from '../../Components/inputs/Checkbox';
 import InputError from '../../Components/inputs/InputError';
 import PrimaryButton from '../../Components/PrimaryButton';
 import { TextInput } from '../../Components/inputs/TextInput';
 import axios from 'axios';
+import { handleLogout } from '@/AuthContext';
+import { BROWSER_URL } from '@/common';
 
 type Inputs = {
     identifier: string;
-    method: string;
     password: string;
+    flow_id: string;
+    challenge: string;
+    csrf_token: string;
 };
 
 export default function LoginForm() {
     const [errorMessage, setErrorMessage] = useState('');
     const [processing, setProcessing] = useState(false);
+    const [user, setUser] = useState<string | null>(null);
     const {
         register,
         handleSubmit,
         formState: { errors }
     } = useForm<Inputs>();
+    const kratosUrl = '/self-service/login/flows?id=';
+    const sessionUrl = '/sessions/whoami';
 
     const submit: SubmitHandler<Inputs> = async (data) => {
         try {
-            let kratosData = {
-                identifier: data.identifier,
-                password: data.password
-            };
-            const urlParams = new URLSearchParams(window.location.search);
-            const login_challenge = urlParams.get('login_challenge');
-            const login_flow = urlParams.get('flow');
-            if (!login_flow) {
-                console.error('No kratos login flow found');
-                window.location.replace('/');
-                return;
-            }
-            const flow_response = await axios.get(
-                '/self-service/login/flows?id=' + login_flow
-            );
+            const attributes = await initFlow();
+            user ? (data.identifier = user) : data.identifier;
+            const reqBody = { ...data, ...attributes };
             setErrorMessage('');
             setProcessing(true);
-            const url = '/self-service/login?flow=' + login_flow;
-            if (login_challenge) {
-                kratosData['method'] = 'oidc';
-            } else {
-                kratosData['method'] = 'password';
-            }
-            const cookie = flow_response.data.ui.nodes[0].attributes.value;
-            kratosData['csrf_token'] = cookie;
-            const response = await axios(url, {
-                method: 'POST',
-                data: kratosData,
-                headers: {
-                    'Content-Type': 'application/json',
-                    'X-CSRF-Token': cookie
+            console.log('Submitting login request', reqBody);
```

Comment:
going to leave these console.logs here until I can get the time to walk either carolina or josh through this entire flow. Not important yet, especially as there are a handlful of other console.logs that need to be removed client side